.github/workflows/pr_typescript.yml

```diff
@@ -0,0 +1,30 @@
+name: Pull request testing
+on:
+  pull_request:
+    branches: [main]
+    paths:
+      # Note that when splitting python vs javascript tests, will update this (once python directory is available & JS is moved into top level dir)
+      - "/typescript/**/*"
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: "0"
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 20
+      - name: Install dependencies
+        run: |
+          yarn install
+      - name: Lint
+        run: |
+          yarn lint
+      - name: Compile
+        run: |
+          yarn compile
+      - name: Test
+        run: |
+          yarn test
```

Comment:
These probably need to have `cd typescript &&` appended to the shell commands?

Reply:
added, seems like its still not triggering the run

Reply:
Maybe remove the "/"? 

In the python one it's `python/**/*`, in typescript it's `/typescript/**/*`

Reply:
that solved it!